### PR TITLE
Handle inputs from stylus

### PR DESCRIPTION
**Pull request by m-kuhn**
Useful for digitizing with a pen

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-69dab07886efeedc0843b0a39c6ae8cbfca5b6f0-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-69dab07886efeedc0843b0a39c6ae8cbfca5b6f0-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-69dab07886efeedc0843b0a39c6ae8cbfca5b6f0-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-69dab07886efeedc0843b0a39c6ae8cbfca5b6f0-x86_64.apk)

**Comment by 3nids** [comment]
Nice!

* it is possible to click on the "+" button with the pen. It will take the last position (very close to the button). I would be in favor of hiding the + button when we are in proximity (HoverHandler?)
* when we are adding a feature, the cursor can be moved, it should be blocked
* missing: unzooming with right click
* in the vertex map tool, as soon as you select a vertex (using the toolbar), the vertex will be moved very far away (since you're close to the border of the screen). We should either:
  * move point on click only (and not on hover)
  * select the vertex to move using the pen and not the toolbar (i.e. prevent from using the toolbar with the pen).
I am not sure which is the most feasible technically.

**Comment by m-kuhn** [comment]
Thanks for the detailed feedback!

> it is possible to click on the "+" button with the pen. It will take the last position (very close to the button). I would be in favor of hiding the + button when we are in proximity (HoverHandler?)

Or hide it anyway? A button that appears when you get close reminds me [of this gui](https://codepen.io/jsvgoncalves/pen/Anlcv)

> when we are adding a feature, the cursor can be moved, it should be blocked

I can't follow, when exactly?

> missing: unzooming with right click

True. I will have a look. I thought about right click + movement -> up for zooming in, down for zooming out, but not sure about the complexity

> 
> * in the vertex map tool, as soon as you select a vertex (using the toolbar), the vertex will be moved very far away (since you're close to the border of the screen). We should either:
>   
>   * move point on click only (and not on hover)
>   * select the vertex to move using the pen and not the toolbar (i.e. prevent from using the toolbar with the pen).
>     I am not sure which is the most feasible technically.

The second one sounds nice. Not sure about complexity either.
I'm not sure about the UX of the first one.

**Comment by 3nids** [comment]
> Or hide it anyway?

When would you hide it? As soon as the pen is on the screen?

>> when we are adding a feature, the cursor can be moved, it should be blocked

> I can't follow, when exactly?

1) click to add
2) the form is open
3) you can still drag the cursor in the map


>> select the vertex to move using the pen and not the toolbar (i.e. prevent from using the toolbar with the pen).

> The second one sounds nice. Not sure about complexity either.

It shouldn't be this bad

1) allow to select the current vertex in the rubberband by clicking in the map (that would be still cool with finger)
2) prevent from using the toolbar with pen (except the remove vertex, cancel and apply buttons)

**Comment by 3nids** [comment]
another issue, when you digitize a polygon, when you press on the validate button, it adds a vertex under the button position.

**Comment by m-kuhn** [comment]
The new version changes a couple of interaction patterns.

- Right click and drag up and down triggers zooming
- Double click finished polygon or line

I think the zooming behavior is more intuitive this way than it was originally specified.

TODO:

- [ ] Verify if pinch is working
- [ ] Hide buttons to add vertex / confirm feature on hover
- [ ] Fix vertex editing in combination with stylus

**Comment by 3nids** [comment]
> Right click and drag up and down triggers zooming

Nice!

> Double click finished polygon or line

What's the reasoning? That this way you can hide all the buttons? Or is it a preference of UX to use the double-clik?

**Comment by m-kuhn** [comment]
> > Double click finished polygon or line
> 
> What's the reasoning? That this way you can hide all the buttons? Or is it a preference of UX to use the double-clik?

Both. For a good UX a right click would be even nicer, but that's already taken for the drag. We would also have long press available (for confirmation or cancel) if we want.

**Comment by 3nids** [comment]
> but that's already taken for the drag

drag != click, no? (I personally hate double clicks)

**Comment by m-kuhn** [comment]
> drag != click, no?

That is correct, but those buttons on styluses are clumsy. It's easy to click (and finish a geometry) when you just actually just wanted to zoom. I'm reluctant to interleave geometry modifiers and navigation.

> (I personally hate double clicks)

I am also not a big fan, but couldn't find a better alternative.

**Comment by 3nids** [comment]
then long press?

**Comment by m-kuhn** [comment]
good idea
It directly closes the form after because of the "mouse release" from the long press. I need to check for a solution.

**Comment by 3nids** [comment]
apparently, long click is not working in the legend (did not tested myself yet)

**Comment by m-kuhn** [comment]
It's slowly getting into shape.

- Long click done
- Selecting vertices in vertex editor done

Open question:

- How to select a feature to edit if in digitize mode? When clicking in digitize mode, we add a vertex.

TODO:

- Confirm coordinate with a second click in vertex editor.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-33fde7be63c7726a6c57126f45d4ee6be08fa3b1-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-33fde7be63c7726a6c57126f45d4ee6be08fa3b1-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-33fde7be63c7726a6c57126f45d4ee6be08fa3b1-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-33fde7be63c7726a6c57126f45d4ee6be08fa3b1-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-eaa3b875d530efdf2d66dc14db7b820694218cc5-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-eaa3b875d530efdf2d66dc14db7b820694218cc5-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-eaa3b875d530efdf2d66dc14db7b820694218cc5-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-eaa3b875d530efdf2d66dc14db7b820694218cc5-x86_64.apk)

**Comment by 3nids** [comment]
> How to select a feature to edit if in digitize mode? When clicking in digitize mode, we add a vertex.

Should we first try to edit and if no feature is found, we create a new one?

We could maybe snap to areas for polygons, snap to segments for lines and obviously vertices for points.
And for polygons and lines, if you snap to vertex you start creating a new feature.

I think it would work well in 99.9999% cases (not to say all) but:
* it brings some complexity to the code
* it might not be very obvious to the user (although, we expect pen users to read the doc before using it?)

**Comment by m-kuhn** [comment]
When we currently identify a feature we do that across all layers (not just the current one). This will make it impossible to digitize a new point on top of an existing polygon.

How about identify on long press?

**Comment by 3nids** [comment]
> When we currently identify a feature we do that across all layers (not just the current one). This will make it impossible to digitize a new point on top of an existing polygon.

correct

> How about identify on long press?

+1

File: src/qml/qgismobileapp.qml [hunk 1]
```diff
@@ -173,9 +192,43 @@ ApplicationWindow {
       onClicked: {
           if (locatorItem.searching) {
               locatorItem.searching = false
-          } else if( !overlayFeatureFormDrawer.visible ) {
-              identifyTool.identify( Qt.point( mouse.x, mouse.y ) )
           }
+          else if (geometryEditorsToolbar.canvasClicked(point)) {
+          }
+          else if ( stateMachine.state === "digitize" && coordinateLocator.sourceLocation !== undefined ) { // the sourceLocation test checks if a (stylus) hover is active
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml]:
```suggestion
          else if ( stateMachine.state === "digitize" && hoverHandler.hovered ) { // the sourceLocation test checks if a (stylus) hover is active
```

File: src/qml/CoordinateLocator.qml [hunk 2]
```diff
@@ -21,6 +21,8 @@ Item {
    */
   property variant overrideLocation: undefined // QgsPoint
 
+  property variant sourceLocation: undefined // Screen coordinate
```
Comment (m-kuhn) [inline src/qml/CoordinateLocator.qml]:
```suggestion
  //! Source location for snapping. If this is set to undefined, the center of the screen will be used.
  //! Overwritten by stylus / hoverHandler.
  property variant sourceLocation: undefined // Screen coordinate
```

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1360d669c79a95fc45da13a9a64960aa68a74cde-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1360d669c79a95fc45da13a9a64960aa68a74cde-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1360d669c79a95fc45da13a9a64960aa68a74cde-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1360d669c79a95fc45da13a9a64960aa68a74cde-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-80094141b60253ec2e7d28382747f89bc8ccfbc4-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-80094141b60253ec2e7d28382747f89bc8ccfbc4-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-80094141b60253ec2e7d28382747f89bc8ccfbc4-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-80094141b60253ec2e7d28382747f89bc8ccfbc4-x86_64.apk)

File: src/qml/MapCanvas.qml [hunk 3]
```diff
@@ -60,106 +67,136 @@ Item {
...
-      mapCanvasWrapper.zoom( pinch.center, pinch.previousScale / pinch.scale )
-      mapCanvasWrapper.pan( pinch.center, pinch.previousCenter )
-      mapArea.panned()
-    }
+          onTriggered: {
+              clicked(firstClickPoint)
+          }
+      }
+
+      onTapCountChanged: {
```
Comment (3nids) [inline src/qml/MapCanvas.qml:85]: @m-kuhn why not using https://doc.qt.io/qt-5/qml-qtquick-taphandler.html#doubleTapped-signal ?

Comment (m-kuhn) [reply src/qml/MapCanvas.qml:85]: Always before a doubleTapped is fired, a single tap is fired as well

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-42c3a51c48b828cdee52d9259fc4075430e72697-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-42c3a51c48b828cdee52d9259fc4075430e72697-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-42c3a51c48b828cdee52d9259fc4075430e72697-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-42c3a51c48b828cdee52d9259fc4075430e72697-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1796329bc49f5b7f491de6f1b7db6a93d6b8300e-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1796329bc49f5b7f491de6f1b7db6a93d6b8300e-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1796329bc49f5b7f491de6f1b7db6a93d6b8300e-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-1796329bc49f5b7f491de6f1b7db6a93d6b8300e-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-7b48c446f67c4eededadd9fa012315f0e016255b-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-7b48c446f67c4eededadd9fa012315f0e016255b-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-7b48c446f67c4eededadd9fa012315f0e016255b-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-7b48c446f67c4eededadd9fa012315f0e016255b-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ac867352e22fd77648edb4f829670dccfa1dd446-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ac867352e22fd77648edb4f829670dccfa1dd446-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ac867352e22fd77648edb4f829670dccfa1dd446-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ac867352e22fd77648edb4f829670dccfa1dd446-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-44e8399ec0d634f349a61cd2e6971ab4e441fe20-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-44e8399ec0d634f349a61cd2e6971ab4e441fe20-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-44e8399ec0d634f349a61cd2e6971ab4e441fe20-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-44e8399ec0d634f349a61cd2e6971ab4e441fe20-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-8ec83d09ead21241898d18d7aba12254da39c729-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-8ec83d09ead21241898d18d7aba12254da39c729-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-8ec83d09ead21241898d18d7aba12254da39c729-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-8ec83d09ead21241898d18d7aba12254da39c729-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-08d4397319327d8c4340c8f433bee5981847d956-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-08d4397319327d8c4340c8f433bee5981847d956-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-08d4397319327d8c4340c8f433bee5981847d956-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-08d4397319327d8c4340c8f433bee5981847d956-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-39b344ae4e08e57ae4fdbe1ee95e15e2e3e2c9bc-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-39b344ae4e08e57ae4fdbe1ee95e15e2e3e2c9bc-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-39b344ae4e08e57ae4fdbe1ee95e15e2e3e2c9bc-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-39b344ae4e08e57ae4fdbe1ee95e15e2e3e2c9bc-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-2cbadaf46b5db8bbfd4ebdc52e3b047fb2263852-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-2cbadaf46b5db8bbfd4ebdc52e3b047fb2263852-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-2cbadaf46b5db8bbfd4ebdc52e3b047fb2263852-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-2cbadaf46b5db8bbfd4ebdc52e3b047fb2263852-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-05a8c682cdb969b9c98c2efb0cc0507e4e35e517-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-05a8c682cdb969b9c98c2efb0cc0507e4e35e517-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-05a8c682cdb969b9c98c2efb0cc0507e4e35e517-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-05a8c682cdb969b9c98c2efb0cc0507e4e35e517-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-0baeee444f44deda4cd3f696d6b8b15ec9c59102-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-0baeee444f44deda4cd3f696d6b8b15ec9c59102-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-0baeee444f44deda4cd3f696d6b8b15ec9c59102-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-0baeee444f44deda4cd3f696d6b8b15ec9c59102-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ba0b7c927982e0f0aae4cf8b8c3bf28591709612-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ba0b7c927982e0f0aae4cf8b8c3bf28591709612-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ba0b7c927982e0f0aae4cf8b8c3bf28591709612-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-863-ba0b7c927982e0f0aae4cf8b8c3bf28591709612-x86_64.apk)

File: src/qml/MapCanvas.qml [hunk 4]
```diff
@@ -60,106 +76,178 @@ Item {
     freeze: false
   }
 
-  PinchArea {
-    id: pinchArea
+//    TapHandler {
```
Comment (nirvn) [inline src/qml/MapCanvas.qml:79]: @3nids , that should probably be taken away prior to merging.

File: src/qml/QuestionDialog.qml [hunk 5]
```diff
@@ -6,7 +6,7 @@ import Theme 1.0
 
 Popup {
     id: questionDialog
-    parent: Overlay.overlay
+    //parent: Overlay.overlay
```
Comment (nirvn) [inline src/qml/QuestionDialog.qml:9]: Just to be safe, maybe re-add parent using ApplicationWindow.overlay.

File: src/qml/MapCanvas.qml (hunk 4 above)
Comment (3nids) [reply src/qml/MapCanvas.qml:79]: this is the code with the confirmed single click which is not used anymore.
But I realized that the double click is doing a zoom in AND an identify at the moment.

You were also suggesting to start the digitizing on a long press (and to finish on long press), @m-kuhn thoughts?

File: src/core/rubberbandmodel.cpp [hunk 6]
```diff
@@ -186,6 +186,12 @@ QgsPoint RubberbandModel::currentCoordinate() const
 
 void RubberbandModel::setCurrentCoordinate( const QgsPoint &currentCoordinate )
 {
+  // play safe, avoid crash
+  if (mPointList.count() == 0)
+    return;
+  // but try to find out
+  Q_ASSERT(mPointList.count() != 0);
```
Comment (nirvn) [inline src/core/rubberbandmodel.cpp:193]: Logic needs fixing.

File: src/core/rubberbandmodel.cpp [hunk 7]
```diff
@@ -237,6 +243,13 @@ void RubberbandModel::addVertex()
   setCurrentCoordinateIndex( mCurrentCoordinateIndex + 1 );
 }
 
+void RubberbandModel::addVertexFromPoint( const QgsPoint &point )
+{
+  qDebug() << "add point " << point.x() << point.y();
```
Comment (3nids) [inline src/core/rubberbandmodel.cpp:248]: remove debug

**Comment by 3nids** [comment]
question pending: 


* currently, short press is for adding feature/vertex when digitizing and identifying when not digitizing. This is confusing. 
  1. since we finalize the digitizing on long press, shall we also start on long press (then short press would always be digitizing)
  2. always identify on long press even when not digitizing
  3. keep as is

1 sounds nice but it's a bit less intuitive.

issues:

* zooming with pen (right-click + vertical drag) does not work
* long press on legend does not work

**Comment by m-kuhn** [comment]
Currently identifying is not possible when digitizing? (I'm not trying to suggest anything, just asking)

**Comment by 3nids** [comment]
Yes identifying is on long press anytime

**Comment by m-kuhn** [comment]
Please correct me but

> currently, short press is for adding feature/vertex when digitizing and identifying when not digitizing. 

and

> Yes identifying is on long press anytime

are contradictory?

**Comment by nirvn** [comment]
For the sake of moving ahead here, should we settle on a behavior? 

If @m-kuhn has reservations on long press to start digitizing, let's stick to current behavior?

**Comment by m-kuhn** [comment]
We discussed this on the phone end of last week. None of the proposals seems to be perfect, so for the sake of moving on the proposal is to stick to what we have and merge. We will ask the customer who uses this in production to give some feedback on usability of this one and adjust later on if it seems to be better.
@3nids ready to merge?

**Comment by nirvn** [comment]
1. merge -> 2. branch off -> 3. party on master :tada:

**Comment by 3nids** [comment]
yes, let's move on :)

**Comment by 3nids** [comment]
merged and branched: https://github.com/opengisch/QField/tree/release-1_6
let the party begin

**Comment by m-kuhn** [comment]
:tada: :taco: :tada:

**Pull request merged**